docs/frontend.asciidoc

```diff
@@ -0,0 +1,108 @@
+to the original source code files for easier debugging.  
+
+[[sourcemap-apply]]
+[float]
+==== How source maps are applied
+
+When source maps have been <<sourcemap-endpoint,uploaded>> and <<frontend-enable,frontend support is enabled>>, 
+source mapping is automatically applied to the `stack trace frames` of all `errors` and `transactions` 
+recorded with the https://github.com/elastic/apm-agent-js-base[javascript frontend agent].
+
+The server tries to find an uploaded source map for every `stack trace frame` of the record.
+The following information is used to find the previously uploaded source map entry:
+
+* the record's `service.name` is matched against the source map's `service_name`
+* the record's `service.version` is matched against the source map's `service_version`
+* the stack trace frame's `abs_path` is matched against the source map's `bundle_filepath`
+
+If multiple source maps with the same meta information are found, 
+the source map with the latest upload timestamp is used. 
+
+In case a matching source map is found and the source map can be applied to the `stack trace frame`, 
+the frame's information is updated with the mapped information before the record is indexed.
+The following information is changed to reflect the original source code file, when source mapping is applied:
+
+* `filename`
+* `function`
+* `line number`
+* `column number`
+* `abs path` is https://golang.org/pkg/path/#Clean[cleaned] to be the shortest path name equivalent to the given path name. 
+
+[[sourcemap-endpoint]]
+[float]
+==== Upload Endpoint 
+To upload a source map you need to send a `HTTP POST` request
+with `Content-Type` set to `multipart/form-data` to the APM Server source maps endpoint:
+
+[source,bash]
+------------------------------------------------------------
+http(s)://{hostname}:{port}/v1/client-side/sourcemaps
+------------------------------------------------------------
+
+[[sourcemap-request-fields]]
+[float]
+===== Request Fields
+The request consists of some meta information and the actual source map.
+The meta information must contain the following fields:
+
+* `service_name`
+* `service_version`
+* `bundle_filepath`
+
+The meta information is used to identify a `source map` when source mapping is applied.
+
+The actual source map must be attached to the request as a `file upload`
+and it must match the specification for 
+https://docs.google.com/document/d/1U1RGAehQwRypUTovF1KRlpiOFze0b-_2gc6fAH0KY0k[Source map revision 3 proposal].
+
+
+[[sourcemap-api-examples]]
+[float]
+==== Example
+
+Send an example source map to the APM Server:
+
+["source","sh",subs="attributes"]
+---------------------------------------------------------------------------
+curl -X POST http://127.0.0.1:8200/v1/client-side/sourcemaps \
+  -F service_name="test-service" 
+  -F service_version="1.0" 
+  -F bundle_filepath="/static/js/main.js.map" 
```

Comment:
"bundle_filepath"="http://example.com/static/js/main.js.map" 

We should also add a note about the "bundle_filepath" needs to be absolute path of the final bundle as it's used in the web application.